uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Hook/BuildForm/ContactImportFieldsFilter.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+class CRM_HRCore_Hook_BuildForm_ContactImportFieldsFilter {
+
+  /**
+   * @var array
+   *  A an array of restricted fields as keys and
+   *  the default values they are to be set to
+   *  as values.
+   */
+  private $restrictedFields = [
+    'dataSource' => CRM_Import_DataSource_CSV::class,
+    'contactType' => CRM_Import_Parser::CONTACT_INDIVIDUAL,
+    'dedupe' => NULL,
+    'fieldSeparator' => ',',
+  ];
+
+  /**
+   * Checks if the form is the contact import form and
+   * performs some logic.
+   *
+   * @param string $formName
+   * @param CRM_Core_Form $form
+   */
+  public function handle($formName, &$form) {
+    if (!$this->shouldHandle($formName)) {
+      return;
+    }
+
+    $this->filterImportFields($form);
+  }
+
+  /**
+   * Freezes the restricted fields and hides them from view and
+   * also sets the default expected values for these fields to
+   * avoid validation issues.
+   *
+   * @param CRM_Core_Form $form
+   */
+  private function filterImportFields($form) {
+    $hiddenOptionsStyle = '#choose-data-source';
+    $defaults = $form->_defaultValues;
+
+    foreach ($this->restrictedFields as $field => $defaultValue) {
+      if ($form->elementExists($field)) {
+        $hiddenOptionsStyle .= ",  .crm-import-datasource-form-block-$field";
+        $form->freeze($field);
+        if ($defaultValue) {
+          $defaults[$field] = $defaultValue;
+        }
+      }
+    }
+
+    $hiddenOptionsStyle .= ' { display: none; }';
```

Comment:
I can see why it needs to be done this way, but it would be so much nicer if Civi had some option like `$form->hide($elementName);`

Reply:
Yeah, I agree, the closest to that is `$form->removeElement($elementName)` but that would totally remove the element which is not what we need.